src/data/__tests__/delete.test.ts

```diff
@@ -0,0 +1,85 @@
+import { deleteVector } from '../delete';
+import {
+  PineconeBadRequestError,
+  PineconeInternalServerError,
+} from '../../errors';
+import { VectorOperationsApi } from '../../pinecone-generated-ts-fetch';
+import type { DeleteOperationRequest } from '../../pinecone-generated-ts-fetch';
+
+describe('deleteVector', () => {
+  test('calls the openapi delete endpoint, passing target namespace', async () => {
+    const mockDelete: (req: DeleteOperationRequest) => Promise<object> =
+      jest.fn();
+    const VOA = { _delete: mockDelete } as VectorOperationsApi;
+    const namespace = 'test-namespace';
+    const ids = ['123', '456', '789'];
+    const deleteAll = false;
+    const filter = {};
```

Comment:
I can probably move these inline, I pulled them out because I'm used to doing that for usability, but it may look cleaner inline.